### ISSUE TITLE
stop-grad on actor on imagine() function.

### ISSUE THREAD

**Issue by tominku** [jsikyoon/dreamer-torch#9]
Hi,

While I'm looking in the imagine() function in models.py,
I wonder why the gradient from the input feature to the action output should be stopped.

(221 line of the models.py)

## detach means the stop-gradient
inp = feat.detach() if self._stop_grad_actor else feat 

action = policy(inp).sample()
succ = dynamics.img_step(state, action, sample=self._config.imag_sample)

You introduced the boolean variable self._stopgrad_actor so that the user can choose to stop it or not. But in my opinion, for the full gradient  from the initial state to the last step of the sequence, shouldn't the feat flow through the computation graph without the stop gradient? I just wonder why there is a stop gradient. have you tried the code without the stop gradient? What was the result like?

I'm struggling to find out the reason for it, and I hope for a good discussion!
Thanks!

**Comment by jsikyoon** [comment]
It is following the way Dreamer v1/2 did. As I remember, Dreamer v1 didn't stop gradients while v2 did because using REINFORCE loss and stop gradient is more stable.

Bests Jaesik.

**Comment by tominku** [comment]
Thanks. I will think more on this.
